state/azure/tablestorage/tablestorage.go

```diff
@@ -0,0 +1,206 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package tablestorage
+
+import (
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/dapr/components-contrib/state"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"strings"
+)
+
+const (
+	fullmetadata        = "application/json;odata=fullmetadata"
+	daprDelim           = "__delim__"
+	valueEntityProperty = "Value"
+
+	accountNameKey = "accountName"
+	accountKeyKey  = "accountKey"
+	tableNameKey   = "tableName"
+)
+
+type StateStore struct {
+	table *storage.Table
+	json  jsoniter.API
+}
+
+type tablesMetadata struct {
+	accountName string
+	accountKey  string
+	tableName   string
+}
+
+// ---- Store interface begin
+
+//initialises connection to table storage, optionally creates a table if it doesn't exist
+func (r *StateStore) Init(metadata state.Metadata) error {
+	meta, err := getTablesMetadata(metadata.Properties)
+	if err != nil {
+		return err
+	}
+
+	client, _ := storage.NewBasicClient(meta.accountName, meta.accountKey)
+	tables := client.GetTableService()
+	r.table = tables.GetTableReference(meta.tableName)
+
+	//check table exists
+	log.Debugf("using table '%s'", meta.tableName)
+	err = r.table.Create(100, fullmetadata, nil)
+	if err != nil {
+		if azureError, ok := err.(storage.AzureStorageServiceError); ok && azureError.Code == "TableAlreadyExists" {
+			//error creating table, but it already exists so we're fine
+			log.Debugf("table already exists")
+		} else {
+			return err
+		}
+	}
+
+	log.Printf("table initialised, account: %s, table: %s", meta.accountName, meta.tableName)
+
+	return nil
+}
+
+func (r *StateStore) Delete(req *state.DeleteRequest) error {
+	log.Debugf("delete %s", req.Key)
+	return r.deleteRow(req)
+}
+
+func (r *StateStore) BulkDelete(req []state.DeleteRequest) error {
+	log.Debugf("bulk delete %v key(s)", len(req))
+	for _, s := range req {
+		err := r.Delete(&s)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *StateStore) Get(req *state.GetRequest) (*state.GetResponse, error) {
+	log.Debugf("fetching %s", req.Key)
+	pk, rk := getPartitionAndRowKey(req.Key)
+	entity := r.table.GetEntityReference(pk, rk)
+	err := entity.Get(1000, fullmetadata, nil)
```

Comment:
What's this magic number(`1000`) ? would you add comment and make it const?

Reply:
request timeout, pushed to constants ;)